plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetViewer.java

```diff
@@ -4038,7 +4038,9 @@ public void refreshWithFilter(DBDDataFilter filter) {
 
     @Override
     public boolean refreshData(@Nullable Runnable onSuccess) {
-        if (!verifyQuerySafety() || !checkForChanges()) {
+        AtomicBoolean panelVisible = new AtomicBoolean(false);
```

Comment:
Use a single-element array instead

Reply:
We use that pattern everywhere. Using an atomic just because it's a referenced type is a dirty hack.